tests/run-make/dump-ice-to-disk/rmake.rs

```diff
@@ -0,0 +1,55 @@
+// This test checks if internal compilation error (ICE) log files work as expected.
+// - Get the number of lines from the log files without any configuration options,
+// then check that the line count doesn't change if the backtrace gets configured to be short
+// or full.
+// - Check that disabling ICE logging results in zero files created.
+// - Check that the ICE files contain some of the expected strings.
+// See https://github.com/rust-lang/rust/pull/108714
+
+use run_make_support::{cwd, fs_wrapper, has_extension, has_prefix, rustc, shallow_find_files};
+
+fn main() {
+    rustc().input("lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
+    let ice_text = get_text_from_ice();
+    let default_set = ice_text.lines().count();
+    let content = ice_text;
+
+    clear_ice_files();
+    rustc().input("lib.rs").env("RUST_BACKTRACE", "short").arg("-Ztreat-err-as-bug=1").run_fail();
+    let short = get_text_from_ice().lines().count();
+    clear_ice_files();
+    rustc().input("lib.rs").env("RUST_BACKTRACE", "full").arg("-Ztreat-err-as-bug=1").run_fail();
+    let full = get_text_from_ice().lines().count();
+    clear_ice_files();
+
+    // The ICE dump is explicitely disabled. Therefore, this should produce no files.
+    rustc().env("RUSTC_ICE", "0").input("lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
+    assert!(get_text_from_ice().is_empty());
```

Comment:
Problem: I'm not very convinced by this assertion, because in `get_text_from_ice()` we concat read content from all discovered ICE files together, meaning that empty ICE output (not a desired ICE output) can be isomorphic to no ICE files? I would expect something more like `shallow_find_files()` on a `rustc-ice-*` pattern for the suitable directory holding true for `is_empty`.

Reply:
Indeed, that is not very precise. Replaced with `shallow_find_files([...]).is_empty()`.